airbyte-integrations/connectors/destination-iceberg-v2/src/main/kotlin/io/airbyte/integrations/destination/iceberg/v2/IcebergTypesComparator.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.iceberg.v2
+
+import org.apache.iceberg.Schema
+import org.apache.iceberg.types.Type
+import org.apache.iceberg.types.Types
+
+/**
+ * Compares two Iceberg [Schema] definitions to identify:
+ * - New columns that do not exist in the "existing" schema.
+ * - Columns whose data types have changed.
+ * - Columns that no longer exist in the incoming schema (i.e., removed).
+ * - Columns that changed from required to optional.
+ */
+class IcebergTypesComparator {
```

Comment:
wondering: which of these is better
* diff existing vs incoming iceberg schema (your current pr)
* diff existing iceberg schema vs incoming AirbyteType (probably dumb)
* convert existing iceberg schema to AirbyteType; diff existing vs incoming AirbyteType (ie. sharable diff logic)

I _think_ diffing iceberg schemas (i.e. keep your current approach) is better? b/c (a) iceberg types don't perfectly align with airbyte types, and (b) iceberg is weird b/c it has recursive schemas?

Reply:
Yup! I did try to make it generic but realised it's not worth trying to solve a future problem when we dont know enough about it. I think for now this being iceberg specific is fair enough and we can re-visit it once we have more understandings of how this can be generalised and made better